pinot-spi/src/main/java/org/apache/pinot/spi/trace/ServerStatsInfo.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.trace;
+
+public class ServerStatsInfo {
+    private final long _submitRequestTimeMs;
+    private final int _requestSentLatencyMs;
+    private final long _receiveDataTableTimeMs;
+    private final int _responseSize;
+    private final int _deserializationTimeMs;
+
+    public ServerStatsInfo(long submitRequestTimeMs, int requestSentLatencyMs,
+        long receiveDataTableTimeMs, int responseSize, int deserializationTimeMs) {
+      _submitRequestTimeMs = submitRequestTimeMs;
+      _requestSentLatencyMs = requestSentLatencyMs;
+      _receiveDataTableTimeMs = receiveDataTableTimeMs;
+      _responseSize = responseSize;
+      _deserializationTimeMs = deserializationTimeMs;
+    }
+
+    public long getSubmitRequestTimeMs() {
```

Comment:
MSE also tracks a bunch of server/stage level stats albeit in a different form. I think we should extend this so it can accept Multistage Query stats too.

cc: @gortiz who worked on adding `MultiStageQueryStats`.

Reply:
That would be great, but unless I'm wrong, all these trace options only support SSE. It is unclear to me how this is being used, and AFAIK StarTree doesn't use this API. `RequestContext` itself seems to be SSE focused given it assumes a single table is involved in the query.